api/v1alpha1/hostedcluster_types.go

```diff
@@ -538,6 +547,111 @@ type IBMCloudPlatformSpec struct {
 	ProviderType configv1.IBMCloudProviderType `json:"providerType,omitempty"`
 }
 
+// PowerVSPlatformSpec defines IBMCloud PowerVS specific settings for components
+type PowerVSPlatformSpec struct {
+	// ResourceGroup is the IBMCloud Resource Group in which the cluster resides. If not
+	// specified then default resource group of the account will be used.
+	//
+	// +immutable
+	ResourceGroup string `json:"resourceGroup,omitempty"`
+
+	// Region is the IBMCloud region in which the cluster resides. This configures the
+	// OCP control plane cloud integrations, and is used by NodePool to resolve
+	// the correct boot image for a given release.
+	//
+	// +immutable
+	Region string `json:"region"`
+
+	// Zone is the availability zone where control plane cloud resources are
+	// created.
+	//
+	// +immutable
+	Zone string `json:"zone,omitempty"`
+
+	// Subnet is the subnet to use for control plane cloud resources.
+	//
+	// +optional
+	Subnet *PowerVSResourceReference `json:"subnet,omitempty"`
```

Comment:
How can this be optional, what happens if it's not set?
Actually I can't see where this is consumed, is it used anywhere?

Reply:
you are right, these can't be optional. This is not used anywhere because we are using Subnet from PowerVSNodePoolPlatform for the machines, but we need this in the future to create new nodepool(will submit a follow up PR)